model/change_point_detection.go

```diff
@@ -328,6 +334,75 @@ func ReplaceChangePoints(ctx context.Context, env cedar.Environment, performance
 	return catcher.Resolve()
 }
 
+func GetTotalPagesForChangePointsGroupedByVersion(ctx context.Context, env cedar.Environment, projectId string, pageSize int) (int, error) {
+	countKey := "count"
+	pipe := appendAfterBaseGetChangePointsByVersionAgg(projectId, bson.M{
+		"$count": countKey,
+	})
+	cur, err := env.GetDB().Collection(perfResultCollection).Aggregate(ctx, pipe)
+	if err != nil {
+		return 0, errors.Wrap(err, "Cannot aggregate to get count of change points grouped by version")
+	}
+	defer cur.Close(ctx)
+	res := struct {
+		Count int `bson:"count"`
+	}{}
+	if cur.Next(ctx) {
+		err = cur.Decode(&res)
+		if err != nil {
+			return 0, errors.Wrap(err, "Cannot decode response of getting count of change points grouped by version")
+		}
+		return int(math.Ceil(float64(res.Count) / float64(pageSize))), nil
+	}
+	return 0, errors.New("Not able to get count of total changepoints matching query")
+}
+
+func appendAfterBaseGetChangePointsByVersionAgg(projectId string, additionalSteps ...bson.M) []bson.M {
+	return append([]bson.M{
+		{
+			"$match": bson.M{
+				bsonutil.GetDottedKeyName(perfInfoKey, perfResultInfoProjectKey):             projectId,
+				bsonutil.GetDottedKeyName(perfAnalysisKey, perfAnalysisChangePointsKey, "0"): bson.M{"$exists": true},
+			},
+		},
+		{
+			"$group": bson.M{
+				"_id":          "$info.version",
+				"perf_results": bson.M{"$push": "$$ROOT"},
+				"version_id":   bson.M{"$first": "$info.version"},
+				"order":        bson.M{"$first": "$info.order"},
```

Comment:
@tychoish question for you: is there any case where these orders are wildly different for a single version? w.r.t. chrono sort.